src/goRunTestCodelens.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import vscode = require('vscode');
+import { CodeLensProvider, TextDocument, CancellationToken, CodeLens, Command } from 'vscode';
+import { getTestFunctions } from './goTest';
+
+export class GoRunTestCodeLensProvider implements CodeLensProvider {
+	public provideCodeLenses(document: TextDocument, token: CancellationToken): CodeLens[] | Thenable<CodeLens[]> {
+		return getTestFunctions(document).then(testFunctions => {
+			return testFunctions.map(func => {
+				let showReferences: Command = {
+					title: 'run test',
+					command: 'go.test.function',
```

Comment:
Instead of creating a new command, I'd prefer re-using the existing `go.test.cursor`. 
You can pass the arguments as `{ functionName: func.name}.
And update `testAtCursor` to look for `args.functionName`. If it exists use it, else use the existing logic to find the test function under the cursor